Cython/Shadow.pyi

```diff
@@ -100,3 +101,28 @@ class typedef(CythonType, Generic[_T]):
 #    def __init__(self) -> None: ...
 
 #def fused_type(*args: Tuple[_T]) -> Type[FusedType[_T]]: ...
+
+def typeof(arg: Any) -> str: ...
+
+_C = TypeVar('_C', bound='Callable')
+
+class _EmptyDecoratorAndManager(object):
+    def __call__(self, x: _C) -> _C: ...
+
+    def __enter__(self) -> None: ...
+
+    def __exit__(
+        self,
+        exc_type: Optional[Type[BaseException]],
+        exc: Optional[BaseException],
+        tb: Optional[TracebackType]
+    ) -> None: ...
+
+cclass = ccall = cfunc = inline = _EmptyDecoratorAndManager()
+
+def declare(
+    t: Optional[Callable[..., _T]] = None,
+    value: Any = ...,
+    *,
+    visibility: str = ...
+) -> _T: ...
```

Comment:
Your current definition does not allow

```python
cython.declare(x=cython.int, y=cython.double)

```

That is why I suggested that you make an overload [here](https://github.com/cython/cython/pull/4335#discussion_r686188937).

Reply:
I forgot about this syntax, is that not when you use it as a decorator?

Reply:
I don't think that `cython.declare` is used as a decorator, if you want here's [the docs](https://cython.readthedocs.io/en/latest/src/tutorial/pure.html#static-typing)

Reply:
I was thinking about `cython.locals()`.